File: easy/704.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length - 1;
        int middle;
        while (l <= r){
            middle = (nums[(l+r) / 2]);
            if (middle == target){
                return (l+r) / 2;
            } else if (target < middle){
                r = ((l+r) / 2) - 1;
            } else {
                l = ((l+r) / 2) + 1;
            }
        }
        return -1;
    }
}

File: easy/704.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        while l <= r:
            i = (r+l) // 2
            if nums[i] == target:
                return i
            elif nums[i] < target:
                l = i + 1
            else:
                r = i - 1
        return -1
            
